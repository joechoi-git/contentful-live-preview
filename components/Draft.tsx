import React from "react";
import { draftMode } from "next/headers";
import { getI18n } from "../locales/server";

export default async function Draft() {
    const { isEnabled } = draftMode();
    const i18n = await getI18n();
    return (
        <div
            id="control-panel"
            className="my-6 border-4 border-accent bg-primary text-primary-content rounded w-full p-4 flex gap-4"
        >
            <p className="font-bold">{i18n("controlPanel")}</p>
            {isEnabled ? (
                <p className="text-primary-content">
                    {i18n("draftEnabled")}{" "}
                    <a href="/api/disable-draft-mode" className="text-primary-content underline">
                        {i18n("disableDraftMode")}
                    </a>
                </p>
            ) : (
                <p className="text-secondary-content">
                    {i18n("draftDisabled")}{" "}
                    <a href="/api/enable-draft-mode" className="text-primary-content underline">
                        {i18n("enableDraftMode")}
                    </a>
                </p>
            )}
        </div>
    );
}
