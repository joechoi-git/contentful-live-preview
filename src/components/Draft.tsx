import React from "react";
import { draftMode } from "next/headers";

export default function Draft() {
    const { isEnabled } = draftMode();
    return (
        <div
            id="control-panel"
            className="my-6 border border-blue-500 rounded w-full p-4 flex gap-4 bg-slate-100"
        >
            <p className="font-bold">Control Panel</p>
            {isEnabled ? (
                <p className="text-green-500">
                    Draft Mode is <strong>Enabled.</strong>{" "}
                    <a href="/api/disable-draft-mode" className="text-blue-500 underline">
                        Disable Draft Mode
                    </a>
                </p>
            ) : (
                <p className="text-red-500">
                    Draft Mode is <strong>Disabled.</strong>{" "}
                    <a href="/api/enable-draft-mode" className="text-blue-500 underline">
                        Enable Draft Mode
                    </a>
                </p>
            )}
        </div>
    );
}
