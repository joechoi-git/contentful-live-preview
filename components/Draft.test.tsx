import { render } from "@testing-library/react";
import Draft from "./Draft";

// Mock the next/headers module
jest.mock("next/headers", () => ({
    draftMode: jest.fn()
}));

// Mock the getI18n function
jest.mock("../locales/server", () => ({
    getI18n: jest.fn()
}));

describe("Draft component", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test("renders draft mode enabled", async () => {
        const { draftMode } = require("next/headers");
        const { getI18n } = require("../locales/server");

        // Mocking draftMode to return enabled status
        draftMode.mockReturnValue({ isEnabled: true });

        // Define the translation keys and values
        const translations: Record<string, string> = {
            controlPanel: "Control Panel",
            draftEnabled: "Draft mode is enabled.",
            disableDraftMode: "Disable draft mode"
        };

        // Mocking getI18n to return a function that returns the expected strings
        getI18n.mockResolvedValue((key: string) => translations[key]);

        const { asFragment } = render(await Draft());

        // Snapshot test
        expect(asFragment()).toMatchSnapshot();
    });

    test("renders draft mode disabled", async () => {
        const { draftMode } = require("next/headers");
        const { getI18n } = require("../locales/server");

        // Mocking draftMode to return disabled status
        draftMode.mockReturnValue({ isEnabled: false });

        // Define the translation keys and values
        const translations: Record<string, string> = {
            controlPanel: "Control Panel",
            draftDisabled: "Draft mode is disabled.",
            enableDraftMode: "Enable draft mode"
        };

        // Mocking getI18n to return a function that returns the expected strings
        getI18n.mockResolvedValue((key: string) => translations[key]);

        const { asFragment } = render(await Draft());

        // Snapshot test
        expect(asFragment()).toMatchSnapshot();
    });
});
