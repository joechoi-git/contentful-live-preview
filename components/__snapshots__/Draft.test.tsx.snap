// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draft component renders draft mode disabled 1`] = `
<DocumentFragment>
  <div
    class="my-6 border-4 border-accent bg-primary text-primary-content rounded w-full p-4 flex gap-4"
    id="control-panel"
  >
    <p
      class="font-bold"
    >
      Control Panel
    </p>
    <p
      class="text-secondary-content"
    >
      Draft mode is disabled. 
      <a
        class="text-primary-content underline"
        href="/api/enable-draft-mode"
      >
        Enable draft mode
      </a>
    </p>
  </div>
</DocumentFragment>
`;

exports[`Draft component renders draft mode enabled 1`] = `
<DocumentFragment>
  <div
    class="my-6 border-4 border-accent bg-primary text-primary-content rounded w-full p-4 flex gap-4"
    id="control-panel"
  >
    <p
      class="font-bold"
    >
      Control Panel
    </p>
    <p
      class="text-primary-content"
    >
      Draft mode is enabled. 
      <a
        class="text-primary-content underline"
        href="/api/disable-draft-mode"
      >
        Disable draft mode
      </a>
    </p>
  </div>
</DocumentFragment>
`;
